encode.go

```diff
@@ -1010,6 +1087,9 @@ func encodeByteString(e *encoderBuffer, em *encMode, v reflect.Value) error {
 	if b := em.encTagBytes(v.Type()); b != nil {
 		e.Write(b)
 	}
+	if vk == reflect.Slice && v.Type().Elem().Kind() == reflect.Uint8 && em.byteSliceEncodingTag != 0 {
+		encodeHead(e, byte(cborTypeTag), em.byteSliceEncodingTag)
+	}
```

Comment:
Tag 21-23 should be encoded before other user registered tags are encoded.  

For example, if `ByteSliceMode` is set to `ByteSliceToByteStringWithExpectedConversionToBase64URL` and `type myBytes []byte` is registered with tag 100, value of `myBytes` type is encoded as `21(100(h'...'))`.

It would be good to add a test for this use case.

Reply:
Great point! We know tags 21-23 apply recursively, so they can go anywhere, but that may not be true for the user-registered tags.